Tasks/DeployVisualStudioTestAgent/TestAgentInstall.ps1

```diff
@@ -40,7 +40,21 @@ function Install-Product($SetupPath, $UserName, $Password, $ProductVersion, $Arg
 
 		if(-not ($exitCode -eq 0 -or $exitCode -eq 3010 -or $exitCode -eq 3015 -or $exitCode -eq 1641))
 		{
-			throw ("The return code {0} was not expected during installation of Test Agent. Please check the installation logs for more details." -f $exitCode.ToString())
+            $testAgentFileExists = Test-Path "$env:SystemDrive\TestAgent\testagent.txt"
+            if($testAgentFileExists)
```

Comment:
When this file gets deleted? Make sure this file deleted on successful installation,  if not all the consecutive installations fail for first time.